tests/forte/data/extractor/feature_test.py

```diff
@@ -0,0 +1,149 @@
+
+    def test_pad(self):
+        self.feature1.pad(4)
+        self.assertEqual(self.feature1.data, [7, 8, 9, 0])
+
+        self.feature1 = self.create_feature1()
+        self.feature1.pad(6)
+        self.assertEqual(self.feature1.data, [7, 8, 9, 0, 0, 0])
+
+        self.feature1 = self.create_feature1(
+            data=[[1, 0, 0], [0, 1, 0]],
+            pad_id=[0, 0, 1],
+            dim=1)
+        self.feature1.pad(4)
+        self.assertEqual(self.feature1.data,
+                         [[1, 0, 0],
+                          [0, 1, 0],
+                          [0, 0, 1],
+                          [0, 0, 1]])
+
+        self.feature2.pad(4)
+        self.assertEqual(self.feature2.get_len(), 4)
+        base_feature_data = [i.data for i in self.feature2.get_sub_features()]
+        self.assertEqual(base_feature_data[:-1],
+                         [[6, 11, 2], [7, 8], [6, 7, 5, 4]])
+        self.assertEqual(len(base_feature_data[-1]), 0)
+
+        self.feature3.pad(4)
+        self.assertEqual(self.feature3.get_len(), 4)
+        base_feature_data = [i.data for i in self.feature3.get_sub_features()]
+        self.assertEqual(base_feature_data[:-2],
+                         [[[0, 1, 0], [1, 0, 0], [1, 0, 0]],
+                          [[1, 0, 0], [0, 1, 0]]])
+        self.assertEqual(len(base_feature_data[-2]), 0)
+        self.assertEqual(len(base_feature_data[-1]), 0)
+
+    def test_unroll(self):
+        self.feature1.pad(4)
+        feature, mask = self.feature1.unroll()
+        self.assertEqual(feature, [7, 8, 9, 0])
+        self.assertEqual(mask, [[1, 1, 1, 0]])
+
+        self.feature2.pad(4)
+        for sub_feature in self.feature2.get_sub_features():
+            sub_feature.pad(4)
+        feature, mask = self.feature2.unroll()
+        self.assertEqual(feature, [[6, 11, 2, 0],
+                                   [7, 8, 0, 0],
+                                   [6, 7, 5, 4],
+                                   [0, 0, 0, 0]])
+        self.assertEqual(mask, [[1, 1, 1, 0],
+                                [[1, 1, 1, 0],
+                                 [1, 1, 0, 0],
+                                 [1, 1, 1, 1],
+                                 [0, 0, 0, 0]]])
+
+        self.feature3.pad(4)
+        for sub_feature in self.feature3.get_sub_features():
+            sub_feature.pad(3)
+        feature, mask = self.feature3.unroll()
+        self.assertEqual(feature, [[[0, 1, 0], [1, 0, 0], [1, 0, 0]],
+                                   [[1, 0, 0], [0, 1, 0], [0, 0, 1]],
+                                   [[0, 0, 1], [0, 0, 1], [0, 0, 1]],
+                                   [[0, 0, 1], [0, 0, 1], [0, 0, 1]]])
+        self.assertEqual(mask, [[1, 1, 0, 0],
+                                [[1, 1, 1],
+                                 [1, 1, 0],
+                                 [0, 0, 0],
+                                 [0, 0, 0]]])
+
+    def create_feature1(self, data=None, pad_id=None, dim=None):
```

Comment:
You can use `@data` to test more data if you want: https://github.com/asyml/forte/blob/master/tests/forte/data/ontology/ontology_code_generator_test.py#L140

Reply:
Thanks! Will look into that.